src/util/mpir_pmi.c

```diff
@@ -167,6 +168,30 @@ const char *MPIR_pmi_job_id(void)
 #endif
 }
 
+int MPIR_pmi_get_parent_port(char *parent_port)
+{
+    int mpi_errno = MPI_SUCCESS;
+    int pmi_errno = PMI_SUCCESS;
+#ifdef USE_PMI1_API
+    pmi_errno = PMI_KVS_Get(pmi_kvs_name, "PARENT_ROOT_PORT_NAME", parent_port,
```

Comment:
The key name -- "PARENT_ROOT_PORT_NAME" -- is defined in `src/mpid/ch4/include/mpidpre.h`, which ultimately determined by the code in `MPID_Init_spawn_multiple`. Therefore, I think this code should be in `mpid_spawn.c`. I'll wrap the `PMI` part into a function `int MPIR_pmi_kvs_get(int src, char *key, char *val, int val_size)` (along with `MPIR_pmi_kvs_put`).